Bottom: 8f2e2e7be2a40612a8094174f3600312aa39f98b
Top:    9c90c117b90afcfb6366389c47f10cee69985cad
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-04-17 07:30:28 +0800

Refresh of tuntap-split-out-xdp

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index b111643..40d2c05 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1643,7 +1643,7 @@ static struct sk_buff *tun_do_xdp(struct tun_struct *tun,
 
 	skb = build_skb(xdp->data_hard_start, buflen);
 	if (!skb) {
-		skb = ERR_PTR(-ENOMEM);
+		err = -ENOMEM;
 		goto err_xdp;
 	}
 
@@ -1654,7 +1654,7 @@ static struct sk_buff *tun_do_xdp(struct tun_struct *tun,
 	return skb;
 err_xdp:
 	put_page(virt_to_head_page(xdp->data_hard_start));
-	return ERR_PTR(-err);
+	return ERR_PTR(err);
 }
 
 static struct sk_buff *tun_build_skb(struct tun_struct *tun,
